moped-editor/src/views/projects/projectView/ProjectComments.js

```diff
@@ -0,0 +1,216 @@
+import React, { useState } from "react";
+import Page from "src/components/Page";
+import {
+  Avatar,
+  Box,
+  Container,
+  Card,
+  CircularProgress,
+  Divider,
+  Grid,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Typography,
+} from "@material-ui/core";
+import AddBoxIcon from "@material-ui/icons/AddBox";
+
+import { makeStyles } from "@material-ui/core/styles";
+import { getSessionDatabaseData } from "src/auth/user";
+import { useQuery, useMutation } from "@apollo/client";
+import { gql } from "apollo-boost";
+import { useParams } from "react-router-dom";
+import parse from "html-react-parser";
+import DOMPurify from "dompurify";
+import ReactQuill from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import ProjectSaveButton from "../newProjectView/ProjectSaveButton";
+
+const COMMENTS_QUERY = gql`
+  query getProjectComments($projectId: Int!) {
+    moped_proj_notes(where: { project_id: { _eq: $projectId } }) {
+      added_by
+      project_note
+      project_id
+      date_created
+      project_note_id
+    }
+  }
+`;
+
+const ADD_PROJECT_COMMENT = gql`
+  mutation AddProjectComment($objects: [moped_proj_notes_insert_input!]!) {
+    insert_moped_proj_notes(objects: $objects) {
+      returning {
+        project_id
+        project_note
+      }
+    }
+  }
+`;
+
+const quillModules = {
+  toolbar: [
+    ["bold", "italic", "underline", "strike"],
+    [{ list: "ordered" }, { list: "bullet" }],
+    ["link"],
+    ["clean"],
+  ],
+};
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: "100%",
+    backgroundColor: theme.palette.background.paper,
+  },
+  commentorText: {
+    display: "inline",
+    fontWeight: "bold",
+  },
+  noteText: {
+    marginTop: theme.spacing(1),
+  },
+  emptyState: {
+    margin: theme.spacing(3),
+  },
+}));
+
+const ProjectComments = () => {
+  const { projectId } = useParams();
+  const classes = useStyles();
+  const userSessionData = getSessionDatabaseData();
+  const [noteText, setNoteText] = useState("");
+  const [commentAddLoading, setCommentAddLoading] = useState(false);
+  const [commentAddSuccess, setCommentAddSuccess] = useState(false);
+
+  const { loading, error, data, refetch } = useQuery(COMMENTS_QUERY, {
+    variables: { projectId },
+  });
+
+  const [addNewComment] = useMutation(ADD_PROJECT_COMMENT, {
+    onCompleted() {
+      setNoteText("");
+      refetch();
+      setCommentAddSuccess(true);
+      setTimeout(() => {
+        setCommentAddLoading(false);
+        setCommentAddSuccess(false);
+      }, 350);
+    },
+  });
```

Comment:
TIL the onCompleted() function 🙌  thank you! Not against setTimeout especially if we want to see an animation.

As a quick side note, mutation functions return a promise, so this could have also been written elsewhere in the code as:

```
addNewComment().then(() => {
     setNoteText("");
      refetch();
      setCommentAddSuccess(true);
...
}).catch(err => { ...});
```